sdmetrics/reports/base_report.py

```diff
@@ -80,10 +80,14 @@ def _validate(self, real_data, synthetic_data, metadata):
                 The metadata of the table.
         """
         self._validate_data_format(real_data, synthetic_data)
-        self._validate_metadata_matches_data(real_data, synthetic_data, metadata)
+        if self.__class__.__name__ == 'DiagnosticReport':
+            table_name = list(metadata.get('tables', {}).keys())
+            if table_name:
+                self.table_names = table_name
```

Comment:
Why do we need to store the table names?

Reply:
To do validations, like using this method: https://github.com/sdv-dev/SDMetrics/blob/c716f20bd036d981f092230200033df0c38799b9/sdmetrics/reports/multi_table/base_multi_table_report.py#L71

Before the table names were instantiated here: https://github.com/sdv-dev/SDMetrics/blob/c716f20bd036d981f092230200033df0c38799b9/sdmetrics/reports/multi_table/base_multi_table_report.py#L61